metron-platform/metron-common/src/test/java/org/apache/metron/common/configuration/ConfigurationsUtilsTest.java

```diff
@@ -18,74 +18,71 @@
 
 package org.apache.metron.common.configuration;
 
-import static org.hamcrest.CoreMatchers.equalTo;
-import static org.junit.Assert.assertThat;
-
-import java.io.IOException;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Optional;
 import org.adrianwalker.multilinestring.Multiline;
-import org.apache.curator.framework.CuratorFramework;
-import org.apache.curator.test.TestingServer;
 import org.apache.metron.TestConstants;
 import org.apache.metron.common.utils.JSONUtils;
-import org.junit.After;
-
+import org.apache.metron.integration.TestZKServer;
 import org.junit.Assert;
-import org.junit.Before;
 import org.junit.Test;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
 
 public class ConfigurationsUtilsTest {
```

Comment:
Can you explain the changes in this utils test? Is this refactoring work, and if so, what is the justification as part of this PR? These tests don't appear to impact or be impacted by the profiler and zk config loading feature as far as I can tell. As a general rule, we allow some degree of refactoring as pertinent to the PR context, but other ancillary changes should typically be split into their own PRs.

Reply:
Good point.  I had originally refactored ConfigurationUtilsTest to break out some zookeeper profile configuration loading / reading code I needed for my own tests.   Subsequently I found an existing API that had the functionality I needed, and modified my code to use that API.  There is no real need for this changes to this module to be in this PR so I have reverted them.

Incidentally - adhoc manual profiling in using Intellij indicates that setting up and tearing down zookeeper for each tests adds about 0.5 - 1 second to the run time for ConfigurationUtilsTest.  Breaking up the large test into 4 smaller tests adds just under 4 seconds to the run time.  Together these two changes pushed run time from around 6 seconds to around 10 seconds.

Reply:
My main concern for the current test setup is that other contributors may leverage that pattern in other places and also when adding additional test cases to the current rigs. So maybe we're at 6 seconds now and up to 10 seconds after this change, but later it's 20, 30, and so on. I think your api idiom is actually pretty good. If you can just split out the parts that actually start/stop the zk component as separate calls, that would be preferable.

I definitely like the config utils test split up. So if you'd be inclined to contribute that still, please do. The review and validation ends up being super simple bc it's entirely confined to the existing Travis test infrastructure.